### PR TITLE
Patch test: Adds test for set on inner array

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

Adds a test based on a github issue. 

Final json after patch:
```json
{
    "id": "084995b1-069b-454c-824c-9a6fbf97353e",
    "taskNum": 43,
    "cost": 1.7976931348623157e+308,
    "pk": "TBD1c35b236-20ff-4b0d-bba8-a5aeb8f355fb",
    "CamelCase": "camelCase",
    "nullableInt": null,
    "valid": true,
    "children": [
        {
            "id": "child1",
            "taskNum": 30,
            "cost": 0,
            "description": "testSet",
            "pk": null,
            "CamelCase": null,
            "nullableInt": null,
            "valid": false,
            "children": null
        },
        {
            "id": "child2",
            "taskNum": 40,
            "cost": 0,
            "description": null,
            "pk": "patched",
            "CamelCase": null,
            "nullableInt": null,
            "valid": false,
            "children": null
        }
    ],
    "_rid": "-+0VAImYn-YBAAAAAAAAAA==",
    "_self": "dbs/-+0VAA==/colls/-+0VAImYn-Y=/docs/-+0VAImYn-YBAAAAAAAAAA==/",
    "_etag": "\"00000000-0000-0000-1e85-1fcf800401d8\"",
    "_attachments": "attachments/",
    "_ts": 1644501012
}
```

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #https://github.com/Azure/azure-cosmos-dotnet-v3/issues/3023

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/DocumentClient.cs [hunk 1]
```diff
@@ -937,7 +937,7 @@ private async Task OpenPrivateAsync(CancellationToken cancellationToken)
 
             // Create the task to start the initialize task
             // Task will be awaited on in the EnsureValidClientAsync
-            Task t = this.initTaskCache.GetAsync(
+            _ = this.initTaskCache.GetAsync(
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/DocumentClient.cs:940]: Is this related to the PR or to fix some warning?

Comment (j82w) [reply Microsoft.Azure.Cosmos/src/DocumentClient.cs:940]: Fix a warning I missed on a previous PR.

**Review by ealsur — APPROVED** [review]
